File: tests/fixtures/sample.java
import java.util.*;

public class sample {
    public static Map<String, Boolean> buildMiniGraph(List<String> nodes) {
        Map<String, Boolean> graph = new HashMap<>();
        for (String node : nodes) {
            graph.put(node, true);
        }
        return graph;
    }

    static class GraphWalker {
        private Map<String, Integer> graph;

        public GraphWalker(Map<String, Integer> graph) {
            this.graph = graph;
        }

        public Set<String> walk(String startId, int depth) {
            Set<String> visited = new HashSet<>();
            Queue<String> queue = new LinkedList<>();
            queue.add(startId);
            while (!queue.isEmpty()) {
                String id = queue.poll();
                if (visited.contains(id) || depth <= 0) continue;
                visited.add(id);
                depth--;
                if (graph.containsKey(id) && graph.get(id) > 0) {
                    queue.add("next");
                }
            }
            return visited;
        }
    }

    public static List<String> lazyBuildContext(String query) {
        String[] terms = query.split(" ");
        List<String> result = new ArrayList<>();
        for (String t : terms) {
            if (t.length() > 2) result.add(t);
        }
        return result;
    }

    static final int MAX_RESULTS = 100;
    static final int DEFAULT_DEPTH = 2;
}

File: tests/fixtures/sample.go
package main

import (
	"fmt"
	"strings"
)

func buildMiniGraph(nodes []string) map[string]bool {
	graph := make(map[string]bool)
	for _, node := range nodes {
		graph[node] = true
	}
	return graph
}

type GraphWalker struct {
	graph map[string]int
}

func (gw *GraphWalker) Walk(startId string, depth int) map[string]bool {
	visited := make(map[string]bool)
	queue := []string{startId}
	for len(queue) > 0 {
		id := queue[0]
		queue = queue[1:]
		if visited[id] || depth <= 0 {
			continue
		}
		visited[id] = true
		depth--
		if gw.graph[id] > 0 {
			queue = append(queue, "next")
		}
	}
	return visited
}

func lazyBuildContext(query string) []string {
	terms := strings.Split(query, " ")
	var result []string
	for _, t := range terms {
		if len(t) > 2 {
			result = append(result, t)
		}
	}
	return result
}

const MAX_RESULTS = 100
const DEFAULT_DEPTH = 2

func main() {
	fmt.Println(buildMiniGraph([]string{"a", "b"}))
}
